### PR TITLE
feat: add pog log steps and pod label assertion step

### PR DESCRIPTION
**Pull request by danielhelfand**
Some of the pod logs steps are difficult to cover with unit tests due to not having way to mock pod logs. If we would like to consider adding e2e tests for kubedog in future, might be a good coverage approach.

Will add docs if changes look good

Steps added:

```
kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) has ([^"]*) in logs since ([^"]*) time and times out in (\S+) seconds`, kdt.KubeContext.ThePodsInNamespaceWithSelectorHasThisSentenceInLogsSinceTime)
	kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) doesn't have ([^"]*) in logs since ([^"]*) time$`, kdt.KubeContext.NoMatchingStringInLogsSinceTime)
	kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) have no errors in logs since ([^"]*) time$`, kdt.KubeContext.ThePodsInNamespaceWithSelectorHaveNoErrorsInLogsSinceTime)
	kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) have some errors in logs since ([^"]*) time$`, kdt.KubeContext.ThePodsInNamespaceWithSelectorHaveSomeErrorsInLogsSinceTime)
	kdt.scenarioContext.Step(`^the pod ([^"]*) in namespace ([^"]*) should have labels ([^"]*)$`, kdt.KubeContext.ThePodInNamespaceShouldHaveLabels)
```

### CODE REVIEW

File: pkg/kubernetes/kube.go [hunk 1]
```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+
+// TODO: Move this when other prs are merged in
+func isRetriable(err error) bool {
+	for _, msg := range retriableErrors {
+		if strings.Contains(err.Error(), msg) {
+			return true
+		}
+	}
+	return false
+}
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:839]: These already exist in `common.go` and `util.go`

File: pkg/kubernetes/kube.go [hunk 2]
```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+
+	return foundCount, nil
+}
+
+func (kc *Client) NoMatchingStringInLogsSinceTime(namespace,
+	selector, searchkeyword, sinceTime string) error {
+
+	if kc.KubeInterface == nil {
+		return errors.Errorf("'Client.KubeInterface' is nil. 'AKubernetesCluster' sets this interface, try calling it before using this method")
+	}
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:881]: Nothing wrong with this currently, but Alfredo's pending PR has a kc.Validate() function that should replace this

File: pkg/kubernetes/kube.go [hunk 3]
```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+
+	since, ok := kc.RememberedTimes[sinceTime]
+	if !ok {
+		return fmt.Errorf("Time '%s' was not remembered", sinceTime)
+	}
+
+	pods, err := listPodsWithLabelSelector(kc.KubeInterface, namespace, selector)
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:888]: use `kc.ListPodsWithLabelSelector`

File: pkg/kubernetes/kube.go [hunk 4]
```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+	}
+	return fmt.Errorf("Pod has %s message in the logs", searchkeyword)
+}
+
+func (kc *Client) ThePodsInNamespaceWithSelectorHaveNoErrorsInLogsSinceTime(namespace string,
+	selector string, sinceTime string) error {
+
+	if kc.KubeInterface == nil {
+		return errors.Errorf("'Client.KubeInterface' is nil. 'AKubernetesCluster' sets this interface, try calling it before using this method")
+	}
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:913]: Replace later with kc.Validate()

File: pkg/kubernetes/kube.go [hunk 5]
```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+
+	since, ok := kc.RememberedTimes[sinceTime]
+	if !ok {
+		return errors.Errorf("Time '%s' was not remembered", sinceTime)
+	}
+
+	pods, err := listPodsWithLabelSelector(kc.KubeInterface, namespace, selector)
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:920]: use kc.ListPodsWithLabelSelector

File: pkg/kubernetes/kube.go [hunk 6]
```diff
@@ -722,6 +747,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+			return errors.Errorf("Pod %s has %d errors", pod.Name, count)
+		}
+	}
+
+	return nil
+}
+
+func (kc *Client) ThePodsInNamespaceWithSelectorHaveSomeErrorsInLogsSinceTime(namespace string,
+	selector string, sinceTime string) error {
+	err := kc.ThePodsInNamespaceWithSelectorHaveNoErrorsInLogsSinceTime(namespace, selector, sinceTime)
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:944]: Isn't this an infinite recursive loop?

Comment (Kyle-Wong) [reply pkg/kubernetes/kube.go:944]: nvm, I see the function names are slightly different

**Comment by danielhelfand** [comment]
> Can you change this to merge into GED-kubedog instead of master?

Yup, can do. My bad.

**Pull request closed without merging**